pkg/master/master.go

```diff
@@ -206,18 +204,19 @@ func (c *Config) createLeaseReconciler() reconcilers.EndpointReconciler {
 	if err != nil {
 		glog.Fatalf("Error creating storage factory: %v", err)
 	}
-	endpointConfig, err := c.ExtraConfig.StorageFactory.NewConfig(kapi.Resource(DefaultStorageEndpoint))
+	endpointPrefix := c.ExtraConfig.StorageFactory.ResourcePrefix(api.Resource("endpoints"))
+	endpointConfig, err := c.ExtraConfig.StorageFactory.NewConfig(kapi.Resource(endpointPrefix))
```

Comment:
`endpointPrefix := c.ExtraConfig.StorageFactory.ResourcePrefix(api.Resource("endpoints"))` looks right

`c.ExtraConfig.StorageFactory.NewConfig(kapi.Resource(endpointPrefix))` definitely does not look right... why are you passing the prefix into `kapi.Resource(endpointPrefix)`?

Reply:
Yes, should remove the kapi.Resource for endpointConfig.

Reply:
the output of `c.ExtraConfig.StorageFactory.ResourcePrefix` is an etcd prefix, the location in etcd where we want to store that resource.

taking that value and making a `schema.GroupResource` out of it doesn't make any sense. this should be 
```
endpointConfig, err := c.ExtraConfig.StorageFactory.NewConfig(kapi.Resource("endpoints"))
```